lib/srv/sess.go

```diff
@@ -669,25 +672,34 @@ func newSession(id rsession.ID, r *SessionRegistry, ctx *ServerContext) (*sessio
 		log: log.WithFields(log.Fields{
 			trace.Component: teleport.Component(teleport.ComponentSession, r.srv.Component()),
 		}),
-		id:              id,
-		registry:        r,
-		parties:         make(map[rsession.ID]*party),
-		participants:    make(map[rsession.ID]*party),
-		login:           ctx.Identity.Login,
-		closeC:          make(chan bool),
-		lingerTTL:       defaults.SessionIdlePeriod,
-		startTime:       startTime,
-		serverCtx:       ctx.srv.Context(),
-		state:           types.SessionState_SessionStatePending,
-		access:          auth.NewSessionAccessEvaluator(policySets, types.SSHSessionKind),
-		scx:             ctx,
-		presenceEnabled: ctx.Identity.Certificate.Extensions[teleport.CertExtensionMFAVerified] != "",
-		io:              NewTermManager(),
-		stateUpdate:     sync.NewCond(&sync.Mutex{}),
-		doneCh:          make(chan struct{}, 2),
-		initiator:       ctx.Identity.TeleportUser,
+		id:                  id,
+		registry:            r,
+		parties:             make(map[rsession.ID]*party),
+		participants:        make(map[rsession.ID]*party),
+		login:               ctx.Identity.Login,
+		closeC:              make(chan bool),
+		lingerTTL:           defaults.SessionIdlePeriod,
+		startTime:           startTime,
+		serverCtx:           ctx.srv.Context(),
+		state:               types.SessionState_SessionStatePending,
+		access:              auth.NewSessionAccessEvaluator(policySets, types.SSHSessionKind),
+		scx:                 ctx,
+		presenceEnabled:     ctx.Identity.Certificate.Extensions[teleport.CertExtensionMFAVerified] != "",
+		io:                  NewTermManager(),
+		stateUpdate:         sync.NewCond(&sync.Mutex{}),
+		doneCh:              make(chan struct{}, 2),
+		initiator:           ctx.Identity.TeleportUser,
+		verboseRequirements: utils.AsBool(ctx.env[teleport.EnvSSHSessionParticipantRequirements]),
 	}
 
+	go func() {
+		<-sess.io.TerminateNotifier()
```

Comment:
From what I can tell, it's not guaranteed that we write to this channel. That means this goroutine can leak. Am I missing something?

Reply:
Correct. Fixed the leak now